src/pkg/services/m365/api/lists.go

```diff
@@ -251,42 +251,49 @@ func (c Lists) PostList(
 		BySiteId(siteID).
 		Lists().
 		Post(ctx, newList, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating list")
+	}
+
+	listItems, err := c.PostListItems(ctx, siteID, restoredList)
+	if err == nil {
+		restoredList.SetItems(listItems)
+		return restoredList, nil
+	}
 
-	return restoredList, graph.Wrap(ctx, err, "creating list").OrNil()
+	// rollback list creation
+	err = c.DeleteList(ctx, siteID, ptr.Val(restoredList.GetId()))
+
+	return nil, graph.Wrap(ctx, err, "deleting restored list after items creation failure").OrNil()
 }
 
-func (c Lists) PostListItem(
+func (c Lists) PostListItems(
 	ctx context.Context,
-	siteID, listID string,
-	oldListByteArray []byte,
+	siteID string,
+	restoredList models.Listable,
```

Comment:
Nit: although it seems like a natural flow to pass in a Listable, it doesn't make sense as an api design.  Since the func only restores list items, it should only take in a slice of list items.  That reduces caller overhead by not needing a Listable, since the Listable isn't actually used except to get its slice of items and ID.
```suggestion
	siteID, listID string,
	items []models.ListItemable,
```